google/cloud/storage/async/client.h

```diff
@@ -561,6 +562,130 @@ class AsyncClient {
+   * `ObjectRewriter`, which the application can use to initiate the copy and to
+   * iterate if the copy requires more than one call to complete.
+   *
+   * @note Application developers should be aware that rewriting large objects
+   *     may take many calls to `AsyncRewriter::Iterate()`. Simple experiments
+   *     rewriting an object within the same bucket or rewriting small objects
+   *     often complete with a single call. For more information, see the
+   *     [Object: rewrite] documentation.
+   *
+   * @param source_bucket_name the name of the bucket containing the source
+   *     object.
+   * @param source_object_name the name of the source object.
+   * @param destination_bucket_name where the destination object will be
+   *     located.
+   * @param destination_object_name what to name the destination object.
+   * @param args a list of optional query parameters and/or request headers.
+   *     Valid types for this operation include `DestinationKmsKeyName`,
+   *      `DestinationPredefinedAcl`, `EncryptionKey`, `IfGenerationMatch`,
+   *      `IfGenerationNotMatch`, `IfMetagenerationMatch`,
+   *      `IfSourceGenerationMatch`, `IfSourceGenerationNotMatch`,
+   *      `IfSourceMetagenerationMatch`, `IfSourceMetagenerationNotMatch`,
+   *      `MaxBytesRewrittenPerCall`, `Projection`, `SourceEncryptionKey`,
+   *      `SourceGeneration`, `UserProject`, and `WithObjectMetadata`.
+   *
+   * @par Idempotency
+   * This operation is purely local, and always succeeds.  The `Iterate()` calls
+   * may call might fail and are only idempotent if restricted by
+   * pre-conditions. It this call the most relevant pre-condition is,
+   * `IfGenerationMatch`.
+   *
+   * @par Example
+   * @snippet storage_async_samples.cc rewrite-object
+   *
+   * @see
+   * [Object: rewrite]:
+   * https://cloud.google.com/storage/docs/json_api/v1/objects/rewrite
+   */
+  template <typename... Args>
+  std::pair<AsyncRewriter, AsyncToken> StartRewrite(
+      std::string source_bucket_name, std::string source_object_name,
+      std::string destination_bucket_name, std::string destination_object_name,
+      Args&&... args) {
+    auto options = SpanOptions(std::forward<Args>(args)...);
+    auto c = connection_->RewriteObject(
+        {RewriteObjectRequest(std::move(source_bucket_name),
+                              std::move(source_object_name),
+                              std::move(destination_bucket_name),
+                              std::move(destination_object_name))
+             .set_multiple_options(std::forward<Args>(args)...),
+         /*.options=*/std::move(options)});
+    auto t = storage_internal::MakeAsyncToken(c.get());
+    return std::make_pair(AsyncRewriter(std::move(c)), std::move(t));
+  }
+
+  /**
+   * Creates an `AsyncRewriter` to copy the source object.
+   *
+   * Applications use this function to reliably copy objects across [location
```

Comment:
same documentation nits as above.

Copy the edits, leaving
- `@param rewrite-token ...`
- `@snippet storage_async_samples.cc resume-rewrite` ?

Reply:
Fixed too.